all-a-board/src/components/Main/NavBar.jsx

```diff
@@ -1,11 +1,41 @@
+import { useEffect } from "react"
 import { useState } from "react"
+import { useSearchParams } from "react-router-dom"
 
 export default function NavBar({children}) {
+    const [searchParams, setSearchParams] = useSearchParams()
+    const initOrder = searchParams.get('order')
     const [isOpen, setIsOpen] = useState(false)
+    const [order, setOrder] = useState(initOrder === 'asc')
+
+
+    function handleSort(e) {
+        const newParams = {sort_by: searchParams.get('sort_by'), order: searchParams.get('order')}
+        if (!newParams.order) delete newParams.order
+        newParams.sort_by = e.target.value
+        setSearchParams(newParams)
+        console.log(JSON.stringify(newParams))
+    }
 
     return <section>
-        <ul><li onClick={() => {setIsOpen(!isOpen)}}>{isOpen? "close" : "Pick A Category"}</li>
+        <ul>{!isOpen && <div>
+            <select onChange={handleSort} value={searchParams.get('sort_by')}>
```

Comment:
```js
            <select onChange={handleSort} value={searchParams.get('sort_by')}>
```

If you don't have a sort_by param then you'll be setting the value of the select to `null` It should be an empty string if you've not set a value for it. (See the console warning for more details)